pkg/integrations/grok_exporter/grok_exporter.go

```diff
@@ -0,0 +1,262 @@
+// Package grok_exporter embeds https://github.com/fstab/grok_exporter
+package grok_exporter //nolint:golint
+
+import (
+	"context"
+	"fmt"
+	v3 "github.com/fstab/grok_exporter/config/v3"
+	"github.com/fstab/grok_exporter/exporter"
+	"github.com/fstab/grok_exporter/tailer"
+	"github.com/fstab/grok_exporter/tailer/fswatcher"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/grafana/agent/pkg/integrations"
+	"github.com/grafana/agent/pkg/integrations/config"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/sirupsen/logrus"
+	"net/http"
+	"os"
+	"time"
+)
+
+var (
+	logfile = "logfile"
+	extra   = "extra"
+)
+
+var additionalFieldDefinitions = map[string]string{
+	logfile: "full path of the logger file",
+	extra:   "full json logger object",
+}
+
+const (
+	numberOfLinesMatchedLabel = "matched"
+	numberOfLinesIgnoredLabel = "ignored"
+	inputTypeWebhook          = "webhook"
+	inputTypeStdin            = "stdin"
+	inputTypeKafka            = "kafka"
+	inputTypeFile             = "file"
+	metricTypeCounter         = "counter"
+	metricTypeGauge           = "gauge"
+	metricTypeHistogram       = "histogram"
+	metricTypeSummary         = "summary"
+)
+
+// Exporter defines the grok_exporter integration.
+type Exporter struct {
+	config          *Config
+	registry        *prometheus.Registry
+	metrics         []exporter.Metric
+	selfMetrics     selfMetrics
+	logTailer       fswatcher.FileTailer
+	retentionTicker *time.Ticker
+	logger          log.Logger
+}
+
+type selfMetrics struct {
+	nLinesTotal                  *prometheus.CounterVec
+	nMatchesByMetric             *prometheus.CounterVec
+	procTimeMicrosecondsByMetric *prometheus.CounterVec
+	nErrorsByMetric              *prometheus.CounterVec
+}
+
+// New creates a new grok_exporter integration
+func New(logger log.Logger, config *Config) (integrations.Integration, error) {
+	metrics, err := config.CreateMetrics()
+	if err != nil {
+		return nil, err
+	}
+
+	registry := prometheus.NewRegistry()
+	for _, m := range metrics {
+		registry.MustRegister(m.Collector())
+	}
+
+	nLinesTotal, nMatchesByMetric, procTimeMicrosecondsByMetric, nErrorsByMetric := initSelfMonitoring(metrics, registry)
+
+	if config.IncludeExporterMetrics {
+		registry.MustRegister(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}))
+		registry.MustRegister(prometheus.NewGoCollector())
```

Comment:
These should already be exposed via the Agent, I'm not sure we need them at the integration-level, too.

Reply:
Checked the metrics endpoint of the agent and you are right, the metrics are present. So removing these.